src/jit/codegenxarch.cpp

```diff
@@ -1528,43 +1514,28 @@ CodeGen::genStructReturn(GenTreePtr treeNode)
     {
         assert(op1->isContained());
 
-        GenTreeLclVarCommon* lclVarPtr = op1->AsLclVarCommon();
-        LclVarDsc* varDsc = &(compiler->lvaTable[lclVarPtr->gtLclNum]);
+        GenTreeLclVarCommon* lclVar = op1->AsLclVarCommon();
+        LclVarDsc* varDsc = &(compiler->lvaTable[lclVar->gtLclNum]);
         assert(varDsc->lvIsMultiRegArgOrRet);
 
-        CORINFO_CLASS_HANDLE typeHnd = varDsc->lvVerTypeInfo.GetClassHandle();
-        assert(typeHnd != nullptr);
-
-        SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
-        compiler->eeGetSystemVAmd64PassStructInRegisterDescriptor(typeHnd, &structDesc);
-        assert(structDesc.passedInRegisters);
-        assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
+        ReturnTypeDesc retTypeDesc;
+        retTypeDesc.Initialize(compiler, varDsc->lvVerTypeInfo.GetClassHandle());
+        assert(retTypeDesc.GetReturnRegCount() == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_RETURN_IN_REGISTERS);
 
-        regNumber retReg0 = REG_NA;
-        unsigned __int8 offset0 = 0;
-        regNumber retReg1 = REG_NA;
-        unsigned __int8 offset1 = 0;
+        var_types type1 = retTypeDesc.GetReturnRegType(1);
+        var_types type2 = retTypeDesc.GetReturnRegType(2);
 
-        var_types type0 = TYP_UNKNOWN;
-        var_types type1 = TYP_UNKNOWN;
+        regNumber reg1 = retTypeDesc.GetABIReturnReg(1);
+        regNumber reg2 = retTypeDesc.GetABIReturnReg(2);
+        assert(reg1 != REG_NA && reg2 != REG_NA);
 
-        getStructTypeOffset(structDesc, &type0, &type1, &offset0, &offset1);
-        getStructReturnRegisters(type0, type1, &retReg0, &retReg1);
-
-        // Move the values into the return registers.
-        // 
-
-        assert(retReg0 != REG_NA && retReg1 != REG_NA);
-
-        getEmitter()->emitIns_R_S(ins_Load(type0), emitTypeSize(type0), retReg0, lclVarPtr->gtLclNum, offset0);
-        getEmitter()->emitIns_R_S(ins_Load(type1), emitTypeSize(type1), retReg1, lclVarPtr->gtLclNum, offset1);
+        // Move the values into the return registers     
+        getEmitter()->emitIns_R_S(ins_Load(type1), emitTypeSize(type1), reg1, lclVar->gtLclNum, 0);
+        getEmitter()->emitIns_R_S(ins_Load(type2), emitTypeSize(type2), reg2, lclVar->gtLclNum, 8);
     }
 
-    // Nothing to do if the op1 of the return statement is a GT_CALL. The call already has the return
-    // registers in the proper return registers. 
-    // This assumes that registers never get spilled. There is an Issue 2966 created to track the need 
-    // for handling the GT_CALL case of two register returns and handle it properly for stress modes 
-    // and potential other changes that may break this assumption.
+    // Assumption: multi-reg return value of a GT_CALL node is never spilled.
+    // TODO: support for multi-reg call nodes.
```

Comment:
You have removed the issue reference - I'm not sure what the protocol is for putting issue numbers in source, but at a minimum this should be a TODO-Bug. Also, since you're touching this code, you could check that the call is not marked GTF_SPILLED. We will already hit the above noway_assert if a GT_RELOAD has been inserted between the call and the return.